class-gf-razorpay.php

```diff
@@ -0,0 +1,255 @@
+                    array(
+                        'type' => 'save',
+                        'messages' => array(
+                            'success' => esc_html__( 'Settings have been updated.', 'razorpay-gravity-forms' )
+                        ),
+                    ),
+                ),
+            ),
+        );
+    }
+
+    public function get_customer_fields_array($feed, $entry)
+    {
+        $fields = array();
+
+        foreach ($this->get_customer_fields() as $field)
+        {
+            $field_id = $feed['meta'][$field['meta_name']];
+
+            $value = rgar($entry, $field_id);
+
+            $fields[$field['meta_name']] = $value;
+        }
+
+        return $fields;
+    }
+
+    public function get_customer_fields() {
+        return array(
+            array( 'name' => 'first_name', 'label' => 'First Name', 'meta_name' => 'billingInformation_firstName' ),
+            array( 'name' => 'last_name', 'label' => 'Last Name', 'meta_name' => 'billingInformation_lastName' ),
+            array( 'name' => 'email', 'label' => 'Email', 'meta_name' => 'billingInformation_email' ),
+            array( 'name' => 'address1', 'label' => 'Address', 'meta_name' => 'billingInformation_address' ),
+            array( 'name' => 'address2', 'label' => 'Address 2', 'meta_name' => 'billingInformation_address2' ),
+            array( 'name' => 'city', 'label' => 'City', 'meta_name' => 'billingInformation_city' ),
+            array( 'name' => 'state', 'label' => 'State', 'meta_name' => 'billingInformation_state' ),
+            array( 'name' => 'zip', 'label' => 'Zip', 'meta_name' => 'billingInformation_zip' ),
+            array( 'name' => 'country', 'label' => 'Country', 'meta_name' => 'billingInformation_country' ),
+        );
+    }
+
+    public function callback() {
+        $entry_id = $_COOKIE['entry_id'];
+
+        $razorpay_order_id = $_COOKIE['razorpay_order_id'];
+
+        $entry = GFAPI::get_entry($entry_id);
+
+        $api = new Api($this->get_plugin_setting('gf_razorpay_key'),
+            $this->get_plugin_setting('gf_razorpay_secret'));
+
+        $attributes = array (
+            'razorpay_order_id'   => $razorpay_order_id,
+            'razorpay_payment_id' => rgpost('razorpay_payment_id'),
+            'razorpay_signature'  => rgpost('razorpay_signature'),
+        );
+
+        $success = false;
+
+        if ($entry  and !empty(rgpost('razorpay_payment_id')) and !empty(rgpost('razorpay_signature')))
+        {
+            try
+            {
+                $api->utility->verifyPaymentSignature($attributes);
+                $success = true;
+            }
+            catch (Errors\SignatureVerificationError $e)
+            {
+                $error = $e->getMessage();
+            }
+        }
+        else
+        {
+            $error = 'Payment Failed';
+        }
+
+        if ($success === true)
+        {
+            $action['id']             = $attributes['razorpay_payment_id'];
```

Comment:
There should be a better way to do this. Please refactor. It's an array, please use array()